packages/wonder-blocks-form/src/components/__tests__/text-area.test.tsx

```diff
@@ -1324,5 +1351,415 @@ describe("TextArea", () => {
+                        />,
+                        defaultOptions,
+                    );
+                    // Act
+                    const field = screen.getByRole("textbox");
+                    await userEvent.type(field, "test");
+
+                    // Assert
+                    expect(field).toHaveAttribute("aria-invalid", "false");
+                });
+
+                it("should have the textarea in an error state after validation fails and the field is blurred", async () => {
+                    // Arrange
+                    render(
+                        <ControlledTextArea
+                            instantValidation={false}
+                            validate={() => "Error message"}
+                        />,
+                        defaultOptions,
+                    );
+                    // Act
+                    const field = screen.getByRole("textbox");
+                    await userEvent.type(field, "test");
+                    await userEvent.tab();
+
+                    // Assert
+                    expect(field).toHaveAttribute("aria-invalid", "true");
+                });
+
+                it("should not be in an error state after a user updates the value after there was an error", async () => {
+                    // Arrange
+                    render(
+                        <ControlledTextArea
+                            validate={() => "Error message"}
+                            instantValidation={false}
+                        />,
+                        defaultOptions,
+                    );
+                    // Act
+                    const field = await screen.findByRole("textbox");
+                    await userEvent.type(field, "t");
+                    // Trigger blur so error is shown
+                    await userEvent.tab();
+                    // Updating the value should clear the error
+                    await userEvent.type(field, "te");
+
+                    // Assert
+                    expect(field).toHaveAttribute("aria-invalid", "false");
+                });
+
+                it("should call onValidate with an empty string when the user changes the value after there was an error", async () => {
+                    // Arrange
+                    const handleValidate = jest.fn();
+                    const errorMsg = "error message";
+                    render(
+                        <ControlledTextArea
+                            validate={() => errorMsg}
+                            onValidate={handleValidate}
+                            instantValidation={false}
+                        />,
+                        defaultOptions,
+                    );
+
+                    // Act
+                    const field = screen.getByRole("textbox");
+                    await userEvent.type(field, "test");
+                    // Blur will trigger error to be shown
+                    await userEvent.tab();
```

Comment:
The test was checking only the last call on the `handleValidate` mock since there were other tests making sure it called it with the error (I was trying to avoid multiple assertions!). I updated the assertion though so it checks for all calls so it includes the first call with the error message. Hope that makes it clearer!

Reply:
This change looks great, thanks!